### ISSUE TITLE
Refactoring main training scripts `train.py` and `default_train.py`

### ISSUE THREAD

**Issue by AxelBogos** [PulkitMadan/convolution-vs-attention#4]
1. Args will be modified to add an argument for each train, val and test dataloaders. The user will be able to provide from which dataset each loader should be from, allowing for easy combinations of any datasets. In turn, DataModules will be initialized and their appropriate data loaders will be fed to the Trainer interface.
2. `define_model` is unchanged
3. A unique timestamp will be created in order to save the run output in a new directory, `run_outputs`. Each run will be uniquely identified by a timestamp, the modelname as well as a log file detailing which combination of train, val test dataloaders was used. `run_outputs` should eventually contain checkpoints, log files and all visualizations relevant to a single run.
4. `default_train` will be modified to include the lightning Trainer interface. This file will be significantly slimmed down as the Trainer interface will abstract away the optimizer, the learning rate scheduler and the logging as simple callbacks. Ideally, all of `train_model_m, model_default_train_m train_model, model_default_train` may be combined into a single abstract training interface receiving as parameters relevent data loaders, run arguments and the model. 
5. function `model_save_load` will be refactored as a utils function
6. WANDB logging will be exclusively implemented as a Trainer interface callback. A new WANDB workspace will be created as the data logged will significantly differ from what we were logging as of now.
